module/zfs/arc.c

```diff
@@ -6254,11 +6255,10 @@ arc_tuning_update(void)
 		arc_c_max = zfs_arc_max;
 		arc_c = arc_c_max;
 		arc_p = (arc_c >> 1);
-		/* Valid range of arc_meta_limit: arc_meta_min - arc_c_max */
-		percent = MIN(zfs_arc_meta_limit_percent, 100);
-		arc_meta_limit = MAX(arc_meta_min, (percent * arc_c_max) / 100);
-		percent = MIN(zfs_arc_dnode_limit_percent, 100);
-		arc_dnode_limit = (percent * arc_meta_limit) / 100;
+		if (arc_meta_limit > arc_c_max)
```

Comment:
(Just checked in Illumos: this `arc_tuning_update()` function doesn't exist there so we're free to go whichever way we like.)